src/Support/OperationExtensions/ResponderExtension.php

```diff
@@ -0,0 +1,149 @@
+        });
+
+        $responses = collect($returnStatements)
+            ->map(function ($returnStatement) {
+                $statusCode = $this->getStatusCode($returnStatement->expr);
+                $transformer = $this->getTransformer($returnStatement->expr);
+
+                if (!$transformer) {
+                    return null;
+                }
+
+                $response = $this->openApiTransformer->toResponse(new Generic($transformer));
+                $response->code = $statusCode ?? 200;
+
+                return $response;
+            })
+            ->filter();
+
+        [$responses, $references] = $responses->partition(fn ($r) => $r instanceof Response);
+
+        $responses = $responses
+            ->groupBy('code')
+            ->map(function (Collection $responses, $code) {
+                if (count($responses) === 1) {
+                    return $responses->first();
+                }
+
+                return Response::make((int) $code)
+                    ->setContent(
+                        'application/json',
+                        Schema::fromType((new AnyOf)->setItems(
+                            $responses->pluck('content.application/json.type')
+                                /*
+                                 * Empty response body can happen, and in case it is going to be grouped
+                                 * by status, it should become an empty string.
+                                 */
+                                ->map(fn ($type) => $type ?: new OpenApiTypes\StringType)
+                                ->all()
+                        ))
+                    );
+            })
+            ->values()
+            ->merge($references)
+            ->all();
+
+        foreach ($responses as $response) {
+            $operation->addResponse($response);
+        }
+    }
+
+    private function usesResponderClass($expression)
+    {
+        if (!$expression) {
+            return false;
+        }
+
+        if ($expression->name->toString() === 'responder') {
+            return true;
+        }
+
+        if (!isset($expression->var)) {
+            return null;
+        }
+
+        return $this->usesResponderClass($expression->var);
+    }
+
+    private function getTransformer($expression)
+    {
+        if (!$expression) {
+            return null;
+        }
+
+        if ($expression->name->toString() === 'success' || $expression->name->toString() === 'error') {
+            if (!isset($expression->args[1])) {
+                return null;
+            }
+
+            if (!$expression->args[1]->value instanceof \PhpParser\Node\Expr\ClassConstFetch) {
+                return null;
+            }
+
+            return $expression->args[1]->value->class->toString();
+        }
+
+        if (!isset($expression->var)) {
+            return null;
+        }
+
+        return $this->getTransformer($expression->var);
+    }
+
+    private function getStatusCode($expression)
```

Comment:
```suggestion
    private function getStatusCode($expression): ?int
```
What's the type of `$expression`?